class-tgm-plugin-activation.php

```diff
@@ -2023,12 +2064,31 @@ public function force_activation() {
 					} elseif ( $this->can_plugin_activate( $slug ) ) {
 						// There we go, activate the plugin.
 						activate_plugin( $plugin['file_path'] );
+						$this->append_transient( 'tgmpa_force_activated_plugins', $slug );
 					}
 				}
 			}
 		}
 
 		/**
+		 * Set/update a transient, appending the new value if the transient already exists.
+		 *
+		 * @since 2.6.0
+		 *
+		 * @param string       $transient_key Name of transient.
+		 * @param string|array $data To set or add to transient.
+		 */
+		protected function append_transient( $transient_key, $data ) {
+			if ( $transient_data = get_transient( $transient_key ) ) {
+				(array) $transient_data[] = $data;
```

Comment:
Actually it isn't, but it is not done correctly in this case ;-)
`get_transient()` might return false if the transient is not found or any type of value if it is found. Making sure we don't get a PHP notice here is a good thing, just should be implemented slightly differently, more along the lines of below.

```php
$transient_data = get_transient( $transient_key );
if ( false === $transient_data ) ) {
	$transient_data = array();
}
if (  ! is_array( $transient_data ) ) {
	$transient_data = (array) $transient_data;
}

if ( ! empty( $data ) ) {
	if ( is_string( $data ) ) {
		$transient_data[] = $data;
	} elseif ( is_array( $data ) ) {
		$transient_data = $transient_data + $data;
	}
}
```